pkg/deployer/helm/ensure.go

```diff
@@ -106,18 +109,56 @@ func (h *Helm) ApplyFiles(ctx context.Context, files map[string]string, exports
 		return err
 	}
 
-	h.DeployItem.Status.Phase = lsv1alpha1.ExecutionPhaseSucceeded
 	h.DeployItem.Status.ProviderStatus = statusData
-	h.DeployItem.Status.ObservedGeneration = h.DeployItem.Generation
 	if err := h.kubeClient.Status().Update(ctx, h.DeployItem); err != nil {
 		h.DeployItem.Status.LastError = lsv1alpha1helper.UpdatedError(h.DeployItem.Status.LastError,
 			currOp, "UpdateStatus", err.Error())
 		return err
 	}
-	h.DeployItem.Status.LastError = nil
+
 	return nil
 }
 
+func (h *Helm) CheckResourcesHealth(ctx context.Context) error {
```

Comment:
can you a some documentation for the function

Reply:
done